File: CS560_Go_Bowling/Game.go
// Game
package main

import (
)

type Game struct{
	sc Scorer
	currentFrame, throwCount int
	
}

func NewGame() Game{
	return Game{NewScorer(), 0, 0}
}

func (game *Game) score() int{
	score := 0
	for frame := 0; frame <= game.currentFrame; frame++{
		score = score + game.sc.returnScore(frame)
	}
	return score
}

func (game *Game) add(pins int){
	game.sc.add(pins, game.throwCount, game.currentFrame)
	if pins == MAX_ROLL && game.currentFrame < (MAX_FRAMES - 1){
		game.throwCount = game.throwCount + 2
	}else{
		game.throwCount++
	}
	
	if game.currentFrame < (MAX_FRAMES - 1) && game.throwCount % 2 == 0{
		game.currentFrame++
		game.throwCount = 0
	}
}

File: BowlingGame2/src/bowling/Game.java
package bowling;

/**
 * Game handles the adding of new throws and gets the current game score by
 * adding the scores from the Scorer object, sc.
 * 
 * @author Max Fowler
 *
 */
public class Game {

	private Scorer sc;
	private int currentFrame;
	private int throwCount;

	/**
	 * Initialize a new game with an empty Scorer, throwCount of 0, and the
	 * starting frame of 0.
	 */
	public Game() {
		sc = new Scorer();
		currentFrame = 0;
		throwCount = 0;
	}

	/**
	 * This method sums, then returns the scores from the Scorer object
	 * 
	 * @return
	 */
	public int score() {
		int score = 0;
		for (int frame = 0; frame <= currentFrame; frame++) {

			score += sc.returnScore(frame);
		}
		return score;
	}

	/**
	 * According to the description, both the Game and the Scorer must handle
	 * pins. Game handles keeping an updated frame and throw count, to ensure
	 * the scorers are placed in their proper array positions.
	 * 
	 * Strikes are represented fair as '10' '0' in the frame array positions.
	 * 
	 * @param pins
	 */
	public void add(int pins) {
		
		if(pins > 10 || pins < 0){
			System.out.println("Pins must be between 0 and 10!");
			return;
		}

		sc.add(pins, throwCount, currentFrame);
		if (pins == Scorer.MAX_ROLL && currentFrame < (Scorer.MAX_FRAMES - 1)) {
			throwCount += 2;
		} else {
			throwCount++;
		}

		if (currentFrame < Scorer.MAX_FRAMES - 1 && throwCount % 2 == 0) {
			currentFrame++;
			throwCount = 0;
		}

	}

}
